src/DependencyInjection/M6WebGuzzleHttpExtension.php

```diff
@@ -213,6 +213,11 @@ protected function getCurlConfig(array $config)
 
     protected function getServiceReference(ContainerBuilder $container, $id)
     {
+        // if $id is already a Reference (e.g. if parts of the config are defined as parameters), return it
+        if ($id instanceof Reference) {
```

Comment:
it's still relevant, because the `$id` type depends on if the user used parameters for `m6web_guzzlehttp` config or not

the following currently work as expected: 

```yml
# config.yml

m6web_guzzlehttp:
    clients:
        default:
            guzzlehttp_cache: 
                 service: '@m6_redis.api'
```

because we don't use "parameters"